packages/jest-config/src/readConfigFileAndSetRootDir.ts

```diff
@@ -94,9 +95,40 @@ const loadTSConfigFile = async (
     });
   } catch (e: any) {
     if (e.code === 'MODULE_NOT_FOUND') {
-      throw new Error(
-        `Jest: 'ts-node' is required for the TypeScript configuration files. Make sure it is installed\nError: ${e.message}`,
-      );
+      let esbuild: typeof build;
+
+      try {
+        esbuild = require('esbuild').build;
+      } catch (error: any) {
+        if (error.code === 'MODULE_NOT_FOUND') {
+          throw new Error(
+            `Jest: 'ts-node' or 'esbuild' is required for the TypeScript configuration files. Make sure it is installed\nError: ${e.message}`,
+          );
+        }
+
+        throw error;
+      }
+
+      const outfile = 'node_modules/jest-config/_jest.config.js';
+
+      await esbuild({
+        // By bundling we add support for importing stuff in the config file.
+        bundle: true,
```

Comment:
I don't think it's a huge risk, but arguably there's a security angle here where if someone were to import a dependency which itself imported a lot of files, the resulting file on disk would be huge.

Reply:
Yeah, I don't think this would be a problem. I added this option only because I was told there will be users who import other files in their config file, then this is a necessity.